pkg/services/govmomi/vcenter/clone.go

```diff
@@ -294,6 +280,17 @@ func Clone(ctx *context.VMContext, bootstrapData []byte, format bootstrapv1.Form
 	disks := devices.SelectByType((*types.VirtualDisk)(nil))
 	isLinkedClone := snapshotRef != nil
 	spec.Location.Disk = getDiskLocators(disks, *datastoreRef, isLinkedClone)
+	spec.Location.Datastore = datastoreRef
+	// Only non-linked clones may expand the size of the template's disk.
+	if snapshotRef == nil {
+		diskSpecs, err := getDiskSpec(ctx, devices)
```

Comment:
@rikatz Sorry tried to figure it out but no idea. How does moving this code block change the outcome?

It seems to me that we would end up with the same deviceSpecs independent of if we run it at the top or here.

What am I missing? :)

Reply:
I would guess something is mutating the devices in between but I have no idea what

Reply:
I think this is a left over change when I was trying to figure out the problem. Previously I was changing the datastore of the disk spec to add the assumed `datastoreRef` and that's why I've moved it.

Let me return to the previous place

Reply:
fixed :) sorry for the left over on it

Reply:
No worries. I was just looking through it and really wondered what I'm missing. I thought everything is possible with something like passing ctx around like we do etc... Just couldn't find it :D